In [59]:
import numpy as np
import bokeh as bk
from bokeh.io import push_notebook
import bokeh.plotting as bkp
import bokeh.models as bkm
from bokeh.io import push_notebook
from ipywidgets import interact
import ipywidgets as widgets
bkp.output_notebook()

Loading BokehJS ...

## Problem 3.

The balanced equation for the reaction between graphite and water is shown below:

$$ H_2O (g) + x O_2 (g) + (1+2 x) C (graphite) = H_2 (g) + (1+2x) CO (g) $$

Calculate the product gas $(H_2$  and  $CO)$ composition at $1000K$. Assume adiabatic conditions(i.e. no heat loss or gain).

Use TSIM to get the equations for CP for each of the components:

####  H20 (g) 298.150----2000.000K :				
	Cp=  6.895 + 2.882*10^(-3)*T + 0.240*10^5*T^(-2)	
$$ 6.895 + 2.882 \cdot 10^{-3}T + 0.240 \cdot 10^5T^{-2} $$

####  O2 (g) 298.150----2000.000K :				
	Cp=  7.230 + 1.006*10^(-3)*T - 0.452*10^5*T^(-2)
$$ 7.230 + 1.006 \cdot 10^{-3}T - 0.452 \cdot 10^5T^{-2} $$
    
#### C (graphite) 298.150----2000.000K :				
	Cp=  3.518 + 1.532*10^(-3)*T - 1.723*10^5*T^(-2)
$$ 3.518 + 1.532 \cdot 10^{-3}T - 1.723 \cdot 10^5T^{-2} $$

#### H2 (g) 298.150----3000.000K :
    Cp=  6.456 + 0.838*10^(-3)*T + 0.165*10^5*T^(-2)
$$ 6.456 + 0.838 \cdot 10^{-3}T +0.165 \cdot 10^5T^{-2} $$   

#### CO (g) 298.150----2000.000K :		
	 Cp=  6.708 + 1.106*10^(-3)*T - 0.062*10^5*T^(-2)	
$$ 6.708 + 1.106 \cdot 10^{-3}T - 0.062 \cdot 10^5T^{-2} $$      

Stoichiometric equation:
$$ H_2O (g) + x O_2 (g) + (1+2 x) C (graphite) = H_2 (g) + (1+2x) CO (g) $$

$$ -57795 + \int_{298.15}^{1000}  6.895 + 2.882 \cdot 10^{-3}T + 0.240 \cdot 10^5T^{-2} + x(7.230 + 1.006 \cdot 10^{-3}T - 0.452 \cdot 10^5T^{-2}) + (1+2x)(3.518 + 1.532 \cdot 10^{-3}T - 1.723 \cdot 10^5T^{-2}) dT = (1+2x)(-26417) + \int_{298.15}^{1000} (1+2x)(6.708 + 1.106 \cdot 10^{-3}T - 0.062 \cdot 10^5T^{-2}) + 6.456 + 0.838 \cdot 10^{-3}T +0.165 \cdot 10^5T^{-2} dT$$

$$ 10949.1x -48824.9 = -16268-42439.5x$$

In [7]:
x=0.44862
result=(1+2*x)/(2+2*x)
print(result)
print(1-result)

0.6548439204208143
0.3451560795791857


##### Answer:
The product gas composition is approximately 65% $CO$, 35 % $H_2$

---

-57797.8 + x(22703/4.184)+(1+2x)(11795/4.184)

H2O formation according to TSIM: -57795
H2O formation according to book: -57797.8

CO formation according to TSIM:-26417
CO formation according to book: -26416.587

H2O according to TSIM: 6208.66
H2O according to book: 6214.15

O2 according to TSIM: 5426.26
O2 according to book: 5426.15

C according to TSIM: 2761.42
C according to book: 2819.1

H2 according to TSIM: 4951.74
H2 according to book: 4942.6

CO according to TSIM: 5197.26
CO according to book: 5184.0


<table>
  <tr>
    <th width="10%"></th>
    <th width="10%">$H_2O F (g)$</th>
    <th width="10%">$CO F (g)$</th>
    <th width="10%">$H_2O (g)$</th>
    <th width="10%">$O_2 (g)$</th>
    <th width="10%">$C (s)$</th>
    <th width="10%">$H_2 (g)$</th>
    <th width="10%">$CO (g)$</th>  
  </tr>
      <th>$TSIM$</th>
      <td>-57795</td>
      <td>-26417</td>
      <td>6208.66</td>
      <td>5426.26</td>
      <td>2761.42</td>
      <td>4951.74</td>
      <td>5197.26</td>
  <tr>
  </tr>
      <th>$Book$</th>
      <td>-57797.8</td>
      <td>-26416.587</td>
      <td>6214.15</td>
      <td>5426.15</td>
      <td>2819.1</td>
      <td>4942.6</td>
      <td>5184.0</td>
  <tr>
</table>|

$$−215826+x⋅22703+(1+2x)⋅11795=20680+(1+2x)⋅−88837$$

$$ -57795 + 6208.66 + x(5426.26) + (1+2x)(2761.42) = 4951.74 + (1+2x)(-26417+5197.26)  $$